[SPARK-32556][INFRA] Fix release script to have urlencoded passwords where required.

ScrapCodes opened the pull request:
### What changes were proposed in this pull request?
1. URL encode the `ASF_PASSWORD` of the release manager.
2. Update the image to install `qpdf` and `jq` dep
3. Increase the JVM HEAM memory for maven build.

### Why are the changes needed?
Release script takes hours to run, and if a single failure happens about somewhere midway, then either one has to get down to manually doing stuff or re run the entire script. (This is my understanding) So, I have made the fixes of a few failures, discovered so far.

1. If the release manager password contains a char, that is not allowed in URL, then it fails the build at the clone spark step.
`git clone "https://$ASF_USERNAME:$ASF_PASSWORD@$ASF_SPARK_REPO" -b $GIT_BRANCH`
               
          ^^^ Fails with bad URL

`ASF_USERNAME` may not be URL encoded, but we need to encode `ASF_PASSWORD`.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
By running the release for branch-2.4, using both type of passwords, i.e. passwords with special chars and without it.

ScrapCodes committed:
Fix release script to have urlencoded passwords where required.

SparkQA commented:
**[Test build #127131 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/127131/testReport)** for PR 29373 at commit [`78d12b3`](https://github.com/apache/spark/commit/78d12b30b49c5ac2b249e34a06d2afcbaa23ec84).

ScrapCodes commented:
@zhengruifeng Hi, it might be useful for you, as you are working on 3.0.1 release.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/31753/
Test PASSed.

SparkQA commented:
**[Test build #127131 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/127131/testReport)** for PR 29373 at commit [`78d12b3`](https://github.com/apache/spark/commit/78d12b30b49c5ac2b249e34a06d2afcbaa23ec84).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/127131/
Test FAILed.

ScrapCodes committed:
increase default maven java heap and code cache

```
[INFO] Spark Integration for Kafka 0.10 ................... SUCCESS [ 57.801 s]
[INFO] Kafka 0.10+ Source for Structured Streaming ........ SUCCESS [01:13 min]
[INFO] Spark Kinesis Integration .......................... SUCCESS [ 59.862 s]
[INFO] Spark Project Examples ............................. SUCCESS [02:10 min]
[INFO] Spark Integration for Kafka 0.10 Assembly .......... SUCCESS [ 45.454 s]
[INFO] Spark Avro ......................................... SUCCESS [01:58 min]
[INFO] Spark Project External Flume Sink .................. SUCCESS [01:34 min]
[INFO] Spark Project External Flume ....................... FAILURE [20:04 min]
[INFO] Spark Project External Flume Assembly .............. SKIPPED
[INFO] Spark Project Kinesis Assembly 2.4.7 ............... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 54:10 min
[INFO] Finished at: 2020-08-06T10:10:12Z
[INFO] ------------------------------------------------------------------------
[ERROR] Java heap space -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/OutOfMemoryError

```

SparkQA commented:
**[Test build #127139 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/127139/testReport)** for PR 29373 at commit [`0edf23e`](https://github.com/apache/spark/commit/0edf23ec92b2f123a73f5d5299b7fa324b1c8c9f).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/31761/
Test PASSed.

SparkQA commented:
**[Test build #127139 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/127139/testReport)** for PR 29373 at commit [`0edf23e`](https://github.com/apache/spark/commit/0edf23ec92b2f123a73f5d5299b7fa324b1c8c9f).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/127139/
Test PASSed.

HyukjinKwon reviewed (approved):
Looks good
